cosmos/scheduler/cmd/scheduler.go

```diff
@@ -31,47 +33,61 @@ type Scheduler struct {
 // Run starts the Cosmos Scheduler
 func (s *Scheduler) Run() {
 	log.Println("starting cosmos scheduler")
-	quit := make(chan bool)
 	overlayUpdates, err := s.watchOverlayUpdates()
 	if err != nil {
 		log.Println(err)
 		return
 	}
+	// safely create or update secret on startup
+	go s.configureIngestionSecret(true)
```

Comment:
I there a reason this is done asynchrously?

Reply:
I was going back and forth on this but ultimately I didn't think it needed to be a blocking operation in this context.

Reply:
Rule of thumb is that things should be synchronous unless there's a real need for them to happen later and out of our sight. Here anything that needs to use the new secret may run too early, creating a race or deadlock (depending), because we don't have a way of knowing when (if) the operation is successful.

Reply:
Ah yeah those are definitely good reasons. Thank you!